### PR TITLE
Fixes kubernetes/kubernetes#28324 Revert ubernetes.io back to k8s-fed…

### PR DESCRIPTION
**Pull request by quinton-hoole**
…eration.com and add a trailing '.'.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml [hunk 1]
```diff
@@ -72,7 +72,7 @@
               export PROJECT="k8s-jkns-e2e-gce-federation"
               export FEDERATION=true
               export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-gce-federation"
-              export FEDERATIONS_DOMAIN_MAP="federation=ubernetes.io"
+              export FEDERATIONS_DOMAIN_MAP="federation=k8s-federation.com"
```
Comment (madhusudancs) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml]: It doesn't hurt to add trailing dot here as well, but we made `KubeDNS` to handle this, so it's fine.

**Comment by madhusudancs** [comment]
LGTM.

I neither have label access nor merge access to this repo, so please feel free to merge.

**Comment by quinton-hoole** [comment]
Thanks @madhusudancs.  @ixdy, @spxtr Could you please grant @madhusudancs access to this repo?  Thanks.

**Comment by ixdy** [comment]
@quinton-hoole done. for future reference, I think anyone can ask to join https://github.com/orgs/kubernetes/teams/test-infra-maintainers.

**Pull request merged**